### ISSUE TITLE
Using unary operator

### ISSUE THREAD

**Issue by jon3r4de** [nus-cs2030/2223-s2#733]
Hi guys, can i check for lab 8 am i allowed to just define a unaryOperator  like below

        UnaryOperator<Num> incrementBySelf = (num) -> num.add(initial);

smth like that, as it does seem like I'm creating a new instance which seems like it violates the below. As I'm currently not sure on how to go about level 3.

There is a single instance property opt of type Optional<T>. This is the only property that you need. As such, you are not allowed to define any other instance properties, including constants, in your implementation.

**Comment by 0liverkong** [comment]
as long as you work solely upon functions defined using succ(), you should be fine.. though using unaryop is unnecessary i think?

**Comment by kyledilao777** [comment]
Hm.... try to work within the specifications of the assignment. can follow what @0liverkong said

**Comment by gene2345** [comment]
yupp i think there is no need for a unary operator in this lab

**Comment by haqanh** [comment]
I think you should not need to do this, as others have pointed out, the functions provided in the AbstractNum class (s, n, valid) are sufficient!

**Comment by Ramanenbh** [comment]
yup like what the others mentioned I only had to use map,flatMap and filter
try manipulating using those, it is more than sufficient.

While you cannot create instance variables, you can create new variables in the function eg: Num temp = Num.zero() which is something that you can exploit

**Comment by kelleyhan25** [comment]
yup, agree with everyone! I don't think I used unary operators at all.

**Comment by AmanDesai34** [comment]
Yup agreed, there shouldnt be a need for unary operators

**Comment by Adem121** [comment]
I pretty much just used loops, succ, equals, flatMap, and defining temp variables within your methods. That's all you need
